drivers/wifi/nrf700x/zephyr/src/zephyr_fw_load.c

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/**
+ * @brief File containing FW load functions for Zephyr.
+ *
+ * Special handling is needed for this file as the patches are stored in
+ * external flash. All RODATA of this file is stored in external flash, so,
+ * any use of RODATA has to be protected by disabling XIP and enabling it
+ * again after use. This means no LOG_* macros (buffered) or buffered printk
+ * can be used in this file, else it will crash.
+ */
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#ifdef CONFIG_NRF_WIFI_PATCHES_EXT_FLASH
+#include <zephyr/drivers/flash/nrf_qspi_nor.h>
+#endif /* CONFIG_NRF_WIFI_PATCHES_EXT_FLASH */
+
+#include <zephyr_fmac_main.h>
+#include <rpu_fw_patches.h>
+
+enum wifi_nrf_status wifi_nrf_fw_load(void *rpu_ctx)
+{
+	enum wifi_nrf_status status = WIFI_NRF_STATUS_FAIL;
+	struct wifi_nrf_fmac_fw_info fw_info;
+#ifdef CONFIG_NRF_WIFI_PATCHES_EXT_FLASH
+	const struct device *flash_dev = DEVICE_DT_GET(DT_INST(0, nordic_qspi_nor));
+#endif /* CONFIG_NRF_WIFI_PATCHES_EXT_FLASH */
+
+	memset(&fw_info, 0, sizeof(fw_info));
+	fw_info.lmac_patch_pri.data = wifi_nrf_lmac_patch_pri_bimg;
+	fw_info.lmac_patch_pri.size = sizeof(wifi_nrf_lmac_patch_pri_bimg);
+	fw_info.lmac_patch_sec.data = wifi_nrf_lmac_patch_sec_bin;
+	fw_info.lmac_patch_sec.size = sizeof(wifi_nrf_lmac_patch_sec_bin);
+	fw_info.umac_patch_pri.data = wifi_nrf_umac_patch_pri_bimg;
+	fw_info.umac_patch_pri.size = sizeof(wifi_nrf_umac_patch_pri_bimg);
+	fw_info.umac_patch_sec.data = wifi_nrf_umac_patch_sec_bin;
+	fw_info.umac_patch_sec.size = sizeof(wifi_nrf_umac_patch_sec_bin);
+
+#ifdef CONFIG_NRF_WIFI_PATCHES_EXT_FLASH
+	nrf_qspi_nor_xip_enable(flash_dev, true);
+#endif /* CONFIG_NRF_WIFI */
+	/* Load the FW patches to the RPU */
+	status = wifi_nrf_fmac_fw_load(rpu_ctx,
+				       &fw_info);
+
+	if (status != WIFI_NRF_STATUS_SUCCESS) {
+		printf("%s: wifi_nrf_fmac_fw_load failed\n", __func__);
+		goto out;
+	}
+out:
+#ifdef CONFIG_NRF_WIFI_PATCHES_EXT_FLASH
+	nrf_qspi_nor_xip_enable(flash_dev, false);
```

Comment:
@anangl do we need a delay here to complete any transactions? In your reference sample there is a sleep of 2secs, but  that will increase the Wi-Fi delay, so, only want to keep minimum delay only if required.

Reply:
No delay is needed. That sleep in the sample is not related to this call.

Reply:
ok, thanks.